File: PHP/8php 1/gooseFilter.php
<?php
/*
|--------------------------------------------------------------------------
| Filter out the geese
|--------------------------------------------------------------------------
|
*/


function gooseFilter($birds) {
   $res = [];
   for ($i=0; $i < count($birds); $i++) { 
      if ($birds[$i] != "African" && $birds[$i] != "Roman Tufted" && $birds[$i] != "Toulouse" && $birds[$i] != "Pilgrim" && $birds[$i] != "Steinbacher") {
         array_push($res, $birds[$i]);
      }
   }
   return $res;
}


function gooseFilter($birds) {
   $geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"];
   return array_values(array_diff($birds, $geese));
}


function gooseFilter($birds): array {
   $geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"];
   return array_values(array_filter($birds, function($bird) use ($geese) {
      return !in_array($bird, $geese);
   }));
}


function gooseFilter($birds) {
   foreach(array_diff($birds, ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]) as $val) $res[] = $val;
   return $res ?: [];  
}
 

print_r(gooseFilter(["Mallard", "Hook Bill", "African", "Crested", "Pilgrim", "Toulouse", "Blue Swedish"]));

File: Python/8py 1/goose_filter.py
"""
|--------------------------------------------------------------------------
| Filter out the geese
|--------------------------------------------------------------------------
|
"""

def goose_filter(birds):
   return list(filter(lambda i: i != "African" and i != "Roman Tufted" and i != "Toulouse" and i != "Pilgrim" and i != "Steinbacher", birds))


geese = {"African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"}
def goose_filter(birds):
    return[x for x in birds if x  not in geese]


def goose_filter(birds):
   return list(filter(lambda x: x not in geese, birds))


def goose_filter(birds):
   for j in geese:    
      for i in birds:    
         if i == j:
            birds.remove(i)
   return birds


print(goose_filter(["Mallard", "Hook Bill", "African", "Crested", "Pilgrim", "Toulouse", "Blue Swedish"]),["Mallard", "Hook Bill", "Crested", "Blue Swedish"])